tasks/az.py

```diff
@@ -0,0 +1,106 @@
+from typing import Optional
+
+from invoke.context import Context
+from invoke.exceptions import Exit
+from invoke.tasks import task
+from pydantic_core._pydantic_core import ValidationError
+
+from . import config, doc, tool
+from .config import get_full_profile_path
+from .deploy import deploy
+from .destroy import destroy
+from .tool import clean_known_hosts as clean_known_hosts_func
+from .tool import get_host, show_connection_message
+
+scenario_name = "az/vm"
+
+
+@task(
+    help={
+        "config_path": doc.config_path,
+        "install_agent": doc.install_agent,
+        "install_updater": doc.install_updater,
+        "agent_version": doc.agent_version,
+        "stack_name": doc.stack_name,
+        "debug": doc.debug,
+        "interactive": doc.interactive,
+        "ssh_user": doc.ssh_user,
+    }
+)
+def create_vm(
+    ctx: Context,
+    config_path: Optional[str] = None,
+    stack_name: Optional[str] = None,
+    install_agent: Optional[bool] = True,
+    install_updater: Optional[bool] = False,
+    agent_version: Optional[str] = None,
+    debug: Optional[bool] = False,
+    interactive: Optional[bool] = True,
+    ssh_user: Optional[str] = None,
+) -> None:
+    """
+    Create a new virtual machine on the cloud.
+    """
+
+    try:
+        cfg = config.get_local_config(config_path)
+    except ValidationError as e:
+        raise Exit(f"Error in config {get_full_profile_path(config_path)}:{e}")
+
+    if not cfg.get_azure().publicKeyPath:
+        raise Exit("The field `azure.publicKeyPath` is required in the config file")
+
+    extra_flags = dict()
+    extra_flags["ddinfra:env"] = "az/agent-sandbox"
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
This could be stored in the `.test-infra-config` file, so that we can switch to `agent-qa`, mostly used internally by maintainers

Reply:
I can do that!

Reply:
Addressed! Thanks